src/defines.rs

```diff
@@ -47,6 +47,19 @@ pub enum AfError {
     ERR_UNKNOWN        = 999
 }
 
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub enum AfBackend {
+    /// Default backend order: OpenCL -> CUDA -> CPU
+    AF_BACKEND_DEFAULT = 0,
+    /// CPU a.k.a sequential algorithms
+    AF_BACKEND_CPU     = 1,
+    /// CUDA Compute Backend
+    AF_BACKEND_CUDA    = 2,
+    /// OpenCL Compute Backend
+    AF_BACKEND_OPENCL  = 3
```

Comment:
This enum's value in ArrayFire is `4`, this is probably the reason OpenCL is not working. ArrayFIre OpenCL libs will be built but i believe it can't be used if this value is not `4` since ArrayFire's native enum has 4 for this backend.